web/src/main/kotlin/com/example/ajaxproject/nats/UserEventNatsService.kt

```diff
@@ -0,0 +1,15 @@
+package com.example.ajaxproject.nats
+
+import com.example.ajax.User
+import com.google.protobuf.GeneratedMessageV3
+import com.google.protobuf.Parser
+import reactor.core.publisher.Flux
+
+interface UserEventNatsService<EventT : GeneratedMessageV3> {
+
+    val parser: Parser<EventT>
+
+    fun subscribeToEvents(userId: String ,eventType: String): Flux<EventT>
```

Comment:
Since your interface is generic, you can subscribe to only one event, so why do you require a client to pass an additional parameter eventType? If I use UserEventNatsService<UserUpdatedEvent> then implementation knows where to subscribe for UserUpdatedEvent